### PR TITLE
Make code action application more async.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**

### CODE REVIEW

**Comment by CyrusNajmabadi** [comment]
Tagging @dotnet/roslyn-ide

**Comment by CyrusNajmabadi** [comment]
@dotnet/roslyn-ide  Simple PR.  Please review tnx.

File: src/EditorFeatures/Core/Implementation/CodeActions/CodeActionEditHandlerService.cs [hunk 1]
```diff
@@ -111,7 +115,7 @@ public ITextBufferAssociatedViewService AssociatedViewService
             {
                 foreach (var document in project.Documents)
                 {
-                    if (!document.HasAnyErrorsAsync(cancellationToken).WaitAndGetResult(cancellationToken))
+                    if (!await document.HasAnyErrorsAsync(cancellationToken).ConfigureAwait(true))
```
Comment (DustinCampbell) [inline src/EditorFeatures/Core/Implementation/CodeActions/CodeActionEditHandlerService.cs]: Simple? `ConfigureAwait(true)` just hurt my brain again. Comment? :smile:

File: src/EditorFeatures/Core/Implementation/CodeActions/CodeActionEditHandlerService.cs [hunk 2]
```diff
@@ -147,7 +151,12 @@ public ITextBufferAssociatedViewService AssociatedViewService
                 // can be undone at once.
                 using (var transaction = workspace.OpenGlobalUndoTransaction(title))
                 {
-                    updatedSolution = ProcessOperations(workspace, fromDocument, title, oldSolution, updatedSolution, operationsList, progressTracker, cancellationToken);
+                    // ConfigureAwait(true) so we come back to the same thread as 
+                    // we do all application on the UI thread.
```
Comment (DustinCampbell) [inline src/EditorFeatures/Core/Implementation/CodeActions/CodeActionEditHandlerService.cs]: Ah. copy this comment up to the other `ConfigureAwait(true)` call site as well.

**Comment by DustinCampbell** [comment]
👍

**Comment by CyrusNajmabadi** [comment]
test vsi please

**Comment by Pilchie** [comment]
:+1:

**Comment by jasonmalinowski** [comment]
:+1:, but we're reasonably confident that the extra pumping on the UI thread isn't bad here?

**Pull request merged**